src/NodeApi/JSValueScopeClosedException.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.JavaScript.NodeApi;
+
+/// <summary>
+/// An exception that was caused by an attempt to access a <see cref="JSValue" /> (or a more
+/// specific JS value type, such as <see cref="JSObject" /> or <see cref="JSArray" />)
+/// after its <see cref="JSValueScope" /> was closed.
+/// </summary>
+public class JSValueScopeClosedException : ObjectDisposedException
+{
+    /// <summary>
+    /// Creates a new instance of <see cref="JSValueDisposedException" /> with an optional
+    /// object name and message.
+    /// </summary>
+    public JSValueScopeClosedException(JSValueScope scope, string? message = null)
+        : base(scope.ScopeType.ToString(), message ?? GetMessage(scope))
+    {
+        Scope = scope;
+    }
+
+    public JSValueScope Scope { get; }
+
+    private static string GetMessage(JSValueScope scope)
+    {
+        return $"The JS value scope of type {scope.ScopeType} was closed.\n" +
```

Comment:
Should we also mention the `JSReference`?

Reply:
Good idea. Added.